tailscale.h

```diff
@@ -0,0 +1,151 @@
+// of tailscale_listen or tailscale_dial on a server.
+//
+// See also: tailscale_up.
+//
+// Returns zero on success or -1 on error, call tailscale_errmsg for details.
+extern int tailscale_start(tailscale sd);
+
+// tailscale_up connects the server to the tailnet and waits for it to be usable.
+//
+// To cancel an in-progress call to tailscale_up, use tailscale_close.
+//
+// Returns zero on success or -1 on error, call tailscale_errmsg for details.
+extern int tailscale_up(tailscale sd);
+
+// tailscale_close shuts down the server.
+//
+// Returns zero on success or -1 on error. No error details are available.
+extern int tailscale_close(tailscale sd);
+
+// The following set tailscale configuration options.
+//
+// Configure these options before any explicit or implicit call to tailscale_start.
+//
+// For details of each value see the godoc for the fields of tsnet.Server.
+//
+// Returns zero on success or -1 on error, call tailscale_errmsg for details.
+extern int tailscale_set_dir(tailscale sd, const char* dir);
+extern int tailscale_set_hostname(tailscale sd, const char* hostname);
+extern int tailscale_set_authkey(tailscale sd, const char* authkey);
+extern int tailscale_set_control_url(tailscale sd, const char* control_url);
+extern int tailscale_set_ephemeral(tailscale sd, int ephemeral);
+// tailscale_set_logfd instructs the tailscale instance to write logs to fd.
+//
+// An fd value of -1 means discard all logging.
+//
+// Returns zero on success or -1 on error, call tailscale_errmsg for details.
+extern int tailscale_set_logfd(tailscale sd, int fd);
+
+// A tailscale_conn is a connection to an address on the tailnet.
+//
+// It is a pipe(2) on which you can use read(2), write(2), and close(2).
+// For extra control over the connection, see the tailscale_conn_* functions.
+typedef int tailscale_conn;
+
+// tailscale_dial connects to the address on the tailnet.
+//
+// The newly allocated connection is written to conn_out.
+//
+// network is a NUL-terminated string of the form "tcp", "udp", etc.
+// addr is a NUL-terminated string of an IP address or domain name.
+//
+// It will start the server if it has not been started yet.
+//
+// Returns zero on success or -1 on error, call tailscale_errmsg for details.
+extern int tailscale_dial(tailscale sd, const char* network, const char* addr, tailscale_conn* conn_out);
+
+// tailscale_conn_localaddr writes the local address into buf.
+//
+// After returning, buf is always NUL-terminated.
+//
+// Returns:
+// 	0      - success
+// 	EBADF  - conn is not a valid tailscale
+// 	ERANGE - insufficient storage for buf
+extern int tailscale_conn_localaddr(tailscale_conn conn, char* buf, char* buflen);
+
+// tailscale_conn_remoteaddr writes the remote address into buf.
+//
+// After returning, buf is always NUL-terminated.
+//
+// Returns:
+// 	0      - success
+// 	EBADF  - conn is not a valid tailscale
+// 	ERANGE - insufficient storage for buf
```

Comment:
Should this have `tailscale_errmsg` for errors? Also `tailscale_conn_localaddr`.